### PR TITLE
feat: Chunhao/ins 4298 slack component dev

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we want more new component in vdp.

This commit

- first stage of slack component

Note
- The unit test has not been done.

### CODE REVIEW

File: pkg/connector/slack/v0/config/definition.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "icon": "assets/xxx.svg",
+  "icon_url": "",
+  "id": "slack",
+  "public": true,
+  "spec": {
+    "connection_specification": {
+      "$schema": "http://json-schema.org/draft-07/schema#",
+      "additionalProperties": false,
+      "properties": {
+        "api_key": {
```
Comment (donch1989) [inline pkg/connector/slack/v0/config/definition.json:17]: Please try to align the naming with the vendor. Slack is using `token`, so let's use `token` here.

**Comment by chuang8511** [comment]
@donch1989 
Do we need this for slack?
It seems we do not need this, so I do not implement.

```
// main.go
func (c connector) Test(sysVars map[string]any, connection *structpb.Struct) error {

	return nil
}


// connector_test.go (refer to other test code)
func TestConnector_Test(t *testing.T) {
```

**Comment by chuang8511** [comment]
Note
- test passed locally, will check why the test failed here.
- Will need to resolve the confliction.
  - [ref](https://instill-ai.slack.com/archives/C016F66FX9Q/p1715249935803859)

**Comment by chuang8511** [comment]
Will comment back the test later and check why it failed in action but not in local.

**Comment by chuang8511** [comment]
Commit to be signed

**Pull request closed without merging**